### PR TITLE
feat(node): upgrade to napi v2

### PR DESCRIPTION
**Pull request by Brooooooklyn**
Merge after napi-rs 2 released

### CODE REVIEW

**Comment by Brooooooklyn** [comment]
@kdy1 I think this one is ready

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you! 
Code changs LGTM.

File: crates/swc/Cargo.toml [hunk 1]
```diff
@@ -80,6 +81,18 @@ swc_plugin_runner = {version = "0.22.0", path = "../swc_plugin_runner", optional
 swc_visit = {version = "0.3.0", path = "../swc_visit"}
 tracing = "0.1.28"
 
+[dependencies.napi-derive]
+version = "2.0.0-beta.1"
```
[the remark this answers is not in the document]
Comment (Brooooooklyn) [reply crates/swc/Cargo.toml]: Reolved

**Review by kdy1 — DISMISSED** [review]
LGTM. Thank you!

**Review by kdy1 — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -56,8 +57,8 @@
         "prepare": "husky install && git config feature.manyFiles true",
         "artifacts": "napi artifacts --dist scripts/npm",
         "prepublishOnly": "tsc -d && napi prepublish -p scripts/npm --tagstyle npm",
-        "build": "tsc -d && napi build --platform --release --cargo-name node --cargo-flags=\"-p node\"",
-        "build:dev": "tsc -d && napi build --platform --cargo-name node --cargo-flags=\"-p node\"",
+        "build": "napi build --platform --cargo-name node --js ./node-swc/src/binding.js --dts ./node-swc/src/binding.d.ts -p node --release && tsc -d",
```
Comment (Brooooooklyn) [inline package.json:60]: @kdy1 oh sorry, the `tsc -d` should be before the `napi build`

Comment (kdy1) [reply package.json:60]: Did it, thank you!

**Pull request merged**